### PR TITLE
Duplicate attributes

### PR DESCRIPTION
**Pull request by fogelito**

### CODE REVIEW

File: src/Database/Database.php [hunk 1]
```diff
@@ -2747,6 +2747,7 @@ public function createIndex(string $collection, string $id, string $type, array
             }
         } catch (DuplicateException $e) {
             // HACK: Metadata should still be updated, can be removed when null tenant collections are supported.
+            // Double check this please
             if (!$this->adapter->getSharedTables()) {
                 throw $e;
             }
```
Comment (abnegate) [inline src/Database/Database.php:2753]: Add setMigrating/isMigrating methods to wrap this

File: src/Database/Validator/Index.php [hunk 2]
```diff
@@ -209,6 +211,21 @@ public function checkIndexLength(Document $index): bool
...
     }
 
+    /**
+     * @param Document $index
+     * @return bool
+     */
+    public function checkReservedNames(Document $index): bool
+    {
+        $key = \strtolower($index->getAttribute('key', $index->getAttribute('$id')));
+        if ($key === 'primary') {
```
Comment (abnegate) [inline src/Database/Validator/Index.php:221]: Let's make an array stored on the adapters $reservedKeys and pass in to the validator

Comment (fogelito) [reply src/Database/Validator/Index.php]: Done!

File: src/Database/Validator/Index.php [hunk 3]
```diff
@@ -84,7 +84,9 @@ public function checkDuplicatedAttributes(Document $index): bool
         $stack = [];
         foreach ($attributes as $key => $attribute) {
             $direction = $orders[$key] ?? 'ASC';
-            $value = \strtolower($attribute . '|' . $direction);
+            // Why did we add direction to duplicates id we check only the current Document?
+            //$value = \strtolower($attribute . '|' . $direction);
```
[the remark this answers is not in the document]
Comment (fogelito) [reply src/Database/Validator/Index.php:95]: Done!

File: src/Database/Database.php (hunk 1 above)
Comment (fogelito) [reply src/Database/Database.php]: Done

**Pull request merged**